lib/rules/static-property-placement.js

```diff
@@ -102,17 +102,17 @@ module.exports = {
 
     /**
      * Check if we should report this property node
-     * @param node
-     * @param expectedRule
+     * @param {ASTNode} node
+     * @param {string} expectedRule
      */
     function reportNodeIncorrectlyPositioned(node, expectedRule) {
       // Detect if this node is an expected property declaration adn return the property name
       const name = classProperties.find(propertyName => {
         if (propertiesToCheck[propertyName](node)) {
-          return propertyName;
+          return !!propertyName;
```

Comment:
Assuming the return value of `reportNodeIncorrectlyPositioned` is always used in a truthy/falsy position, this just returns `true` instead of "truthy"…